stub_uploader/get_version.py

```diff
@@ -123,19 +124,25 @@ def compute_stub_version(
     return new_version
 
 
+# `today` can be passed as an argument for testing purposes.
 def find_unused_date(
-    max_published: Version, base_version_changed: bool, date: datetime.date
+    max_published: Version, base_version_changed: bool, today: datetime.date
 ) -> datetime.date:
+    # Is not necessarily a date in case the package predates the switch
+    # to the current versioning scheme.
+    last_date = max_published.release[-1]
+
     # If the base version changed, we should always be able to use the current date.
     if base_version_changed:
-        return date
+        return today
 
     # Make sure that the last published release is not unreasonable large.
-    assert int(date.strftime("%Y%m%d")) + 14 > max_published.release[-1]
+    assert int(today.strftime("%Y%m%d")) + 14 > last_date
```

Comment:
Where does 14 come from here? Is it just arbitrary?

Reply:
Yes. It's very unlikely we publish more than 14 versions in a day.

Reply:
While 13 versions are just a little unlikely, I guess :)

Reply:
> While 13 versions are just a little unlikely, I guess :)

a little unlikely, but very unlucky